File: swimming-contest-software/Model/src/main/java/domain/dtos/RaceDTO.java
package domain.dtos;

import domain.enums.SwimmingDistance;
import domain.enums.SwimmingStyle;

import java.io.Serializable;

public class RaceDTO implements Serializable {
    private final SwimmingDistance swimmingDistance;
    private final SwimmingStyle swimmingStyle;
    private final Integer noSwimmers;

    public RaceDTO(SwimmingDistance swimmingDistance, SwimmingStyle swimmingStyle, Integer noSwimmers) {
        this.swimmingDistance = swimmingDistance;
        this.swimmingStyle = swimmingStyle;
        this.noSwimmers = noSwimmers;
    }

    public SwimmingDistance getDistance() {
        return swimmingDistance;
    }

    public SwimmingStyle getStyle() {
        return swimmingStyle;
    }

    public Integer getNoSwimmers() {
        return noSwimmers;
    }
}

File: Model/Domain/DTOs/RaceDTO.cs
using System;
using Model.Domain.Enums;

namespace Model.Domain.DTOs;

[Serializable]
public class RaceDTO
{
    public RaceDTO(SwimmingDistance swimmingDistance, SwimmingStyle swimmingStyle, int noSwimmers)
    {
        SwimmingDistance = swimmingDistance;
        SwimmingStyle = swimmingStyle;
        NoSwimmers = noSwimmers;
    }

    public SwimmingDistance SwimmingDistance { get; }
    public SwimmingStyle SwimmingStyle { get; }
    public int NoSwimmers { get; }
}